pkg/arvo/sys/vane/clay.hoon

```diff
@@ -2488,20 +2489,24 @@
         ::  candidate list.
         ::
         =.  bases
-          =/  new-reachable  (reachable-takos:ze i.takos)
-          (~(put in (~(dif in bases) new-reachable)) i.takos)
-        outer-loop(takos t.takos)
+          =/  new-reachable  (reachable-takos:ze tako)
+          (~(put in (~(dif in bases) new-reachable)) tako)
+        outer-loop
       ::  If it's an ancestor of another candidate, this is not most
       ::  recent, so skip and try next in worklist.
       ::
       =/  base-reachable  (reachable-takos:ze i.base-list)
-      ?:  (~(has in base-reachable) i.takos)
-        outer-loop(takos t.takos)
+      ?:  (~(has in base-reachable) tako)
+        outer-loop
       bases-loop(base-list t.base-list)
     ::  Append parents to list and recurse
     ::
-    =/  bob-yaki  (~(got by hut.ran) i.takos)
-    outer-loop(takos (weld t.takos (skip p.bob-yaki ~(has in done))))
+    =/  bob-yaki  (~(got by hut.ran) tako)
+    =/  new-candidates  (skip p.bob-yaki ~(has in done))
+    %_  outer-loop
+      done   (~(gas in done) new-candidates)
```

Comment:
If I'm reading this right, `done` is somewhat misleadingly named, and it's really more like the set of commits we've marked as being relevant.  The old code didn't add to `done` here, but if my understanding is correct, then this should be a performance improvement and not break functionality.

Reply:
You're right, `done` should be named `seen` or `enqueued` or something.  As the comment specifies, it's the ones we've already checked OR are already in our list to check.  If we don't care about the order, we could make `takos` (our worklist) a set and check against that directly, but I think we're best off processing the worklist in breadth-first order, which is what the queue does.